tests/test_modules.py

```diff
@@ -87,18 +92,28 @@ def test_collection(self):
         # show "settarg" collection
         cmd.describe("settarg")
 
-class TestModuleCollection:
+    @pytest.mark.xfail(reason="Collection Name must be string when saving", raises=TypeError)
+    def test_type_error_save_collection(self):
+        cmd = Module(["settarg"])
+        cmd.save(1)
 
```

Comment:
Just to clarify the usage here - when we do "save" or "describe" the idea is that we are saving a collection?

And a second question - when I create a Module instance, since I can provide a list, is the idea that I'm actually creating multiple modules? Or it's sets of software that would be defined under one Module on a system?